src/Renderer/c3DEngine.js

```diff
@@ -141,21 +145,68 @@ c3DEngine.prototype.getRenderer = function getRenderer() {
     return this.renderer;
 };
 
-c3DEngine.prototype.renderViewTobuffer = function renderViewTobuffer(view, buffer, x, y, width, height) {
-    // TODO Deallocate render texture
+/**
+ * Render view to a Uint8Array.
+ *
+ * @param {View} view - The view to render
+ * @param {object} [zone] - partial zone to render (zone x/y uses view coordinates)
+ * @return {THREE.RenderTarget} - Uint8Array, 4 bytes per pixel. The first pixel in
+ * the array is the bottom-left pixel.
+ */
+c3DEngine.prototype.renderViewToBuffer = function renderViewToBuffer(view, zone) {
+    if (!zone) {
+        zone = {
+            x: 0,
+            y: 0,
+            width: this.width,
+            height: this.height,
+        };
+    }
+
+    this.renderViewToRenderTarget(view, this.fullSizeRenderTarget, zone);
+
+    const pixelBuffer = new Uint8Array(4 * zone.width * zone.height);
+    this.renderer.readRenderTargetPixels(
+        this.fullSizeRenderTarget,
+        zone.x, this.height - (zone.y + zone.height), zone.width, zone.height, pixelBuffer);
+
+    return pixelBuffer;
+};
+
+/**
+ * Render view to a THREE.RenderTarget.
+ *
+ * @param {View} view - The view to render
+ * @param {THREE.RenderTarget} [target] - destination render target. Default value: full size render target owned by c3DEngine.
+ * @param {object} [zone] - partial zone to render (zone x/y uses view coordinates) Note: target must contain complete zone
+ * @return {THREE.RenderTarget} - the destination render target
+ */
+c3DEngine.prototype.renderViewToRenderTarget = function renderViewToRenderTarget(view, target, zone) {
+    if (!target) {
+        target = this.fullSizeRenderTarget;
+    }
     const current = this.renderer.getRenderTarget();
-    this.renderer.setRenderTarget(buffer);
-    this.renderer.setViewport(0, 0, buffer.width, buffer.height);
-    this.renderer.setScissor(x, y, width, height);
-    this.renderer.setScissorTest(true);
-    this.renderer.clearTarget(buffer, true, true, false);
-    this.renderer.render(view.scene, view.camera.camera3D, buffer);
-    this.renderer.setScissorTest(false);
-    var pixelBuffer = new Uint8Array(4 * width * height);
-    this.renderer.readRenderTargetPixels(buffer, x, y, width, height, pixelBuffer);
+
+    // Don't use setViewport / setScissor after setting the renderTarget, because
+    // they affect non-rendertarget rendering as well.
+    // Instead set properties before calling setRenderTarget
```

Comment:
Not sure I understand this well, but I suggest:

> Don't use setViewport / setScissor on renderer because they would affect screen rendering as well. Instead set them on the render target

because the problem is not the call order (as I thought first because of `after setting...`) but the fact they are set on the renderer instead of the renderTarget. But that's assuming I understand this well...

Reply:
Yep, done.